src/modules/NtpClient.rb

```diff
@@ -1099,10 +1099,12 @@ def update_cfa_record(record)
     def write_ntp_conf
       records_for_write.each do |record|
         unless record["cfa_record"]
-          record["cfa_record"] = CFA::NtpConf::Record.record_class(record["type"]).new
-          ntp_conf.records << record["cfa_record"]
+          ntp_conf.records << CFA::NtpConf::Record.record_class(record["type"]).new
+          record["cfa_record"] = ntp_conf.records.last
```

Comment:
We fixed an obscure bug here but the method remains untested? Will it break again once we touch CFA?